### PR TITLE
r_reg_get_bytes: Return NULL if there are no bytes in the arena for the given regset type

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

This pr fixes the following error from the dbg.stepback test under asan, in a more proper manner (https://github.com/radareorg/radare2/runs/1096331101#step:13:362):

arena_null

It appears that there are no bytes because the regset overlaps with another regset. Also, if one regset is missing, it should not abort the entire reg writing process.

**Test plan**

The above error should not appear in the asan build's dbg.stepback test output. All other builds are green.

**Closing issues**

...

### CODE REVIEW

File: libr/reg/arena.c [hunk 1]
```diff
@@ -43,7 +43,11 @@ R_API ut8 *r_reg_get_bytes(RReg *reg, int type, int *size) {
 		*size = sz;
 	}
 	buf = malloc (sz);
-	if (buf && reg->regset[type].arena->bytes) { // TODO change 2nd check to something more reasonable
+	if (!reg->regset[type].arena->bytes) { // DBG
+		eprintf ("sz %d\n", sz);
+		*(ut8 *)0 = 0;
```
Comment (trufae) [inline libr/reg/arena.c:48]: use r_sys_breakpoint() for debugging :P

Comment (trufae) [reply libr/reg/arena.c:48]: or just use an assert

**Review by trufae — COMMENTED** [review]
,

File: libr/reg/arena.c (hunk 1 above)
Comment (kazarmy) [reply libr/reg/arena.c:48]: i'll use r_sys_breakpoint()/an assert after this pr

**Comment by kazarmy** [comment]
[Description](https://github.com/radareorg/radare2/pull/17657#issue-487231707) updated.

**Comment by kazarmy** [comment]
[Description](https://github.com/radareorg/radare2/pull/17657#issue-487231707) updated again.

**Review by trufae — APPROVED** [review]

**Pull request merged**